drivers/serial/uart_stm32.c

```diff
@@ -1490,38 +1490,32 @@ static int uart_stm32_pm_control(const struct device *dev,
 #endif /* CONFIG_PM_DEVICE */
 
 #ifdef CONFIG_UART_ASYNC_API
-#define DMA_CHANNEL_CONFIG(id, dir)					\
-	DT_INST_DMAS_CELL_BY_NAME(id, dir, channel_config)
-#define DMA_FEATURES(id, dir)						\
-	DT_INST_DMAS_CELL_BY_NAME(id, dir, features)
-#define DMA_CTLR(id, dir)						\
-	DT_INST_DMAS_CTLR_BY_NAME(id, dir)
 
 /* src_dev and dest_dev should be 'MEMORY' or 'PERIPHERAL'. */
 #define UART_DMA_CHANNEL_INIT(index, dir, dir_cap, src_dev, dest_dev)	\
-	.dma_dev = DEVICE_DT_GET(DMA_CTLR(index, dir)),			\
+	.dma_dev = DEVICE_DT_GET(STM32_DMA_CTLR(index, dir)),			\
 	.dma_channel = DT_INST_DMAS_CELL_BY_NAME(index, dir, channel),	\
 	.dma_cfg = {							\
 		.dma_slot = DT_INST_DMAS_CELL_BY_NAME(index, dir, slot),\
```

Comment:
`STM32_DMA_SLOT`

Reply:
shouldn't the .dma_dev be made by the STM32_DMA_CTLR ?

Reply:
Sure